includes/Stories_Renderer/Renderer.php

```diff
@@ -0,0 +1,512 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories\Stories_Renderer;
+
+use Google\Web_Stories\Interfaces\Renderer as RenderingInterface;
+use Google\Web_Stories\Model\Story;
+use Google\Web_Stories\Story_Query as Stories;
+use Google\Web_Stories\Story_Post_Type;
+use Google\Web_Stories\Traits\Assets;
+use Iterator;
+
+/**
+ * Renderer class.
+ *
+ * @SuppressWarnings(PHPMD.ExcessiveClassComplexity)
+ * @SuppressWarnings(PHPMD.TooManyPublicMethods)
+ * @implements Iterator<int, \WP_Post>
+ */
+abstract class Renderer implements RenderingInterface, Iterator {
+
+	use Assets;
+
+	/**
+	 * Web Stories stylesheet handle.
+	 *
+	 * @var string
+	 */
+	const STYLE_HANDLE = 'web-stories-list-styles';
+
+	/**
+	 * Stories object
+	 *
+	 * @var Stories Stories object
+	 */
+	protected $stories;
+
+	/**
+	 * Story attributes
+	 *
+	 * @var array An array of story attributes.
+	 */
+	protected $attributes = [];
+
+	/**
+	 * Story posts.
+	 *
+	 * @var array An array of story posts.
+	 */
+	protected $story_posts = [];
+
+	/**
+	 * Pointer to iterate over stories.
+	 *
+	 * @var int
+	 */
+	private $position = 0;
+
+	/**
+	 * Height for displaying story.
+	 *
+	 * @var string
+	 */
+	protected $height = '430';
+
+	/**
+	 * Width for displaying story.
+	 *
+	 * @var string
+	 */
+	protected $width = '285';
+
+	/**
+	 * Whether content overlay is enabled for story.
+	 *
+	 * @var bool
+	 */
+	protected $content_overlay;
+
+	/**
+	 * Constructor
+	 *
+	 * @param Stories $stories Stories instance.
+	 */
+	public function __construct( Stories $stories ) {
+
+		$this->stories    = $stories;
+		$this->attributes = $this->stories->get_story_attributes();
+	}
+
+	/**
+	 * Output markup for amp stories.
+	 *
+	 * @return string
+	 */
+	abstract public function render();
+
+	/**
+	 * Retrieve current story.
+	 *
+	 * @return mixed|void
+	 */
+	public function current() {
+		return $this->story_posts[ $this->position ];
+	}
+
+	/**
+	 * Retrieve next story.
+	 *
+	 * @retrun void
+	 */
+	public function next() {
+		++ $this->position;
+	}
+
+	/**
+	 * Retrieve the key for current node in list.
+	 *
+	 * @return bool|float|int|string|void|null
+	 */
+	public function key() {
+		return $this->position;
+	}
+
+	/**
+	 * Check if current position is valid.
+	 *
+	 * @return bool|void
+	 */
+	public function valid() {
+		return isset( $this->story_posts[ $this->position ] );
+	}
+
+	/**
+	 * Reset pointer to start of the list.
+	 *
+	 * @return void
+	 */
+	public function rewind() {
+		$this->position = 0;
+	}
+
+	/**
+	 * Perform initial setup for object.
+	 *
+	 * @return void
+	 */
+	public function init() {
+
+		add_filter( 'web_stories_get_stories_posts', [ $this, 'prepare_story_modal' ] );
+		$this->story_posts = $this->stories->get_stories();
+		remove_filter( 'web_stories_get_stories_posts', [ $this, 'prepare_story_modal' ] );
```

Comment:
That's an odd API. The `get_stories()` method should just do the transformations and return `Story` objects directly.